### PR TITLE
chore: Bump spring boot to 3.3.4

### PR DESCRIPTION
**Pull request by fhennig**
### Summary

Update spring 3.3.3 -> 3.3.4 and fix a bug that came up.

### PR Checklist

- ~All necessary documentation has been adapted.~
- ~The implemented feature is covered by an appropriate test.~

### CODE REVIEW

**Comment by fhennig** [comment]
error introduced by this commit: https://github.com/spring-projects/spring-boot/commit/04891746ff1280c3ddf88d1fd16028c274198d34

File: backend/src/test/kotlin/org/loculus/backend/SpringBootTestWithoutDatabase.kt [hunk 1]
```diff
@@ -24,5 +25,11 @@ annotation class SpringBootTestWithoutDatabase
 class TestConfig {
     @Bean
     @Primary
-    fun dataSource(): DataSource = mockk()
+    fun dataSource(): DataSource = mockk<DataSource>()
+        .also {
+            // fix for: https://github.com/loculus-project/loculus/pull/3064
```
Comment (fengelniederhammer) [inline backend/src/test/kotlin/org/loculus/backend/SpringBootTestWithoutDatabase.kt:30]: I think I would rather link to https://github.com/spring-projects/spring-boot/issues/42323

File: backend/src/test/kotlin/org/loculus/backend/SpringBootTestWithoutDatabase.kt [hunk 2]
```diff
@@ -24,5 +25,11 @@ annotation class SpringBootTestWithoutDatabase
...
+            every {
+                it.isWrapperFor(eq(org.springframework.jdbc.datasource.lookup.AbstractRoutingDataSource::class.java))
```
Comment (fengelniederhammer) [inline backend/src/test/kotlin/org/loculus/backend/SpringBootTestWithoutDatabase.kt:32]: We could import `AbstractRoutingDataSource`

**Review by fengelniederhammer — APPROVED** [review]
LGTM :+1:

File: backend/src/test/kotlin/org/loculus/backend/SpringBootTestWithoutDatabase.kt (hunk 1 above)
Comment (fhennig) [reply backend/src/test/kotlin/org/loculus/backend/SpringBootTestWithoutDatabase.kt:30]: sure

File: backend/src/test/kotlin/org/loculus/backend/SpringBootTestWithoutDatabase.kt (hunk 2 above)
Comment (fhennig) [reply backend/src/test/kotlin/org/loculus/backend/SpringBootTestWithoutDatabase.kt]: done

**Pull request merged**